### ISSUE TITLE
User with permissions from group cannot edit a manufacturer

### ISSUE THREAD

**Issue by mtausig** [snipe/snipe-it#6502]
#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**

I have a group set up in my snipe-it instances, which has full rights on manufacturers (view/create/edit/delete). But when one such user tries to edit a manufacturer, the Sad Panda and a 403 error appears.

**To Reproduce**

1. Create  a group with the following permissions: Reports, Assets (View, Checkin, Checkout, Audit, View requestable), Accessories (View, Edit, Checkin, Checkout), Consumables (View, Checkout), Components (View), Users (View, Edit), Categories/Departments/Status Labels/Custom Fields/Suppliers/Locations/Companies (View), Manufacturers (All), Self (2FA)
2. Login as a user which is a member of that group
3. Open Settings -> Manufacturers
4. Click *Edit* on any manufacturer
5. 403 error appears

**Expected behavior**

The edit form should open.

**Server (please complete the following information):**
 - Snipe-IT Version: v4.6.6
 - OS: Ubuntu 16.04
 - Web Server: nginx
 - PHP Version 7.0.32

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Browser Firefox and Chromium
 - Version 63/71

**Error Messages**

The *Gate* tab in the debug bar shows one entry which is an error:

```
array:4 [
  "ability" => "edit"
  "result" => false
  "user" => 6831
  "arguments" => "[0 => App\Models\Manufacturer]"
]
```
The *Auth* tab shows the following:

```
web    array:2 [ "name" => "myusername" "user" => array:34 [ "id" => 6831 ...
api      array:2 [ "name" => "Guest" "user" => array:1 [ "guest" => true ] ]
```
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
- If a stacktrace is provided in the error, include that too: **None**
- Any errors that appear in your browser's error console: **None**
- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo: **cannot alter group membership in the demo**
- Include any additional information you can find in `storage/logs` and your webserver's logs: **None**

**Additional context**

- Is this a fresh install or an upgrade?: **Upgraded from v4.5.0**
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc): **git**
- Indicate whether or not you've manually edited any data directly in the database: **no**

**Comment by JonathonReinhart** [comment]
I'm also seeing this on a fresh v4.6.7 install. Here's the group settings:

![image](https://user-images.githubusercontent.com/1916566/51207423-bb8b9380-18d8-11e9-8381-f593ba2da268.png)

But the user in that group (with all permissions set to Inherit) gets a 403 when attempting to access `/manufacturers/2/edit`.

**Comment by stbc** [comment]
Hi! Same with categories over here! All rights granted via group but 403 when trying to edit an existing categorie. Same user with same group can add categories nevertheless

**Comment by DeusMaximus** [comment]
I'm also seeing this as well on 4.6.6. If I make the group a blanket admin, it'll inherit properly, but if I remove admin access and try to define custom permissions in their group, the users in that group don't seem to pick up the custom permissions I give them.

Will try to update to 4.6.13 and see if this is still occurring.

**Comment by stbc** [comment]
Yes, not fixed but present.

**Comment by Sxderp** [comment]
Seeing this issue too. 4.7.4

**Comment by mtausig** [comment]
Still relevant.
Informationen zum Datenschutz: www.fh-campuswien.ac.at/datenschutzerklaerung

**Comment by telecosrv** [comment]
Seeing this issue too. v4.7.6 build 4143. Hope it can be fixed!

**Comment by neilcxy** [comment]
问题影响supplier,categories,Manufacturers.找到app\http\controllers 对应的3个文件，把edit 修改为 update 可以解决。
$this->authorize('edit',supplier::class)  change to
$this->authorize('update',supplier::class)

do the same operation as the other 2 files.

**Comment by AngelusDulcis** [comment]
Anyone got a line on which files were being edited here? I'm experiencing the same issue as well.

**Comment by stbc** [comment]
Translate puts out the following

The problem affects suppliers, categories, Manufacturers. Find the 3 files corresponding to app\http\controllers, and modify edit to update to solve.

**Comment by dhiart92** [comment]
seeing to on Version v4.7.7 - build 4160 (master)

> 问题影响supplier,categories,Manufacturers.找到app\http\controllers 对应的3个文件，把edit 修改为 update 可以解决。
> $this->authorize('edit',supplier::class) change to
> $this->authorize('update',supplier::class)
> 
> do the same operation as the other 2 files.

problem solved for that but an error when clicking "save" update (category, manufacturer & suppliers)

**Comment by dhiart92** [comment]
I think the problem is with this form of permissions, does anyone know the location of the script for that?

![image](https://user-images.githubusercontent.com/50138737/68081980-26e32980-fe49-11e9-8492-2884217f2105.png)

**Comment by hapm** [comment]
This will be fixed with #7493 .

**Comment by dhiart92** [comment]
> This will be fixed with #7493 .

ok solved
thanks @hapm

**Comment by nenad-sladojevic** [comment]
Same issue is still present in v4.9.2 with Suppliers when corresponding group permissions are:
![image](https://user-images.githubusercontent.com/20811423/91835638-ee1f0700-ec49-11ea-9551-54eea8d5ad0b.png)

In which Snipe-IT version mentioned fix (#7493) is applied ?
I failed to find it on:
https://github.com/snipe/snipe-it/releases

**Comment by hapm** [comment]
The reason is it never hit the master branch or v4. yet and is still only merged on develop and branch integrations/rc5-4-2020-08-31-dev-into-master. It will be included in v5 as you can see by the [diff](https://github.com/snipe/snipe-it/compare/v4.9.4...v5.0.0-beta-3.0#diff-f24828b5618cc25602cf2e51653f3afe).

**Comment by hapm** [comment]
Sorry for my deleted comment, I missread yours @nenad-sladojevic . The permissions for suppliers should have been fixed in 48bbbe0f40f2f619ec459bacb3e7bd42df0f259f and should be on master. The pull request was #5807 and didn't reference this issue. It has been fixed in milestone v4.6.0. Sure your issue is related? Do you get the sad panda when trying to edit a supplier? Probably best option is to create a new issue and reference this one if you think these are related.

**Comment by tberanek** [comment]
@hapm Looks like it was merged into the develop branch but never made it's way into master.
